Makefile

```diff
@@ -81,7 +81,7 @@ clean: ## Clean
 # note: to review coverage execute: go tool cover -html=/tmp/coverage.out
 .PHONY: test
 test: validate_go ## Test
-	go test -p 1 -race -covermode=atomic -coverprofile=/tmp/coverage.out ./...
+	go test -p 1 -race -coverpkg=./... -covermode=atomic -coverprofile=/tmp/coverage.out ./...
```

Comment:
@eranra please note that I added the `-coverpkg` flag.
I added it because I noticed that the code coverage didn't increase for `aggregate.go` although this PR tests execution paths that were not tested by the other tests. The reason is that the test in this PR is in a different go package from `aggregate.go`. Adding the flag solves this. The following repo helped me understand this:
https://github.com/yudai/go_cover